server/src/main/java/com/google/musicanalysis/site/RequestBuilder.java

```diff
@@ -0,0 +1,78 @@
+package com.google.musicanalysis.site;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+/*
+    Handles gathering data from the Musixmatch API.
+    This is structured the way it is becuase I will be adding more
+    to each function.
+*/
+public class RequestBuilder {
+
+  // Url building
+  private String baseUrl = "https://api.musixmatch.com/ws/1.1/";
+  private String operation;
+  private String outputReqs = "?format=jsonp&callback=callback&";
+  private String parameters;
+  private final String apiKey = "&apikey=";
+
+  private String response;
+
+  /**
+   * Constructor.
+   *
+   * @param operation the API method to send to Musixmatch
+   * @param parameters the parameter string to be sent to Musixmatch
+   */
+  public RequestBuilder(String operation, String parameters) {
+    this.operation = operation;
+    this.parameters = parameters;
+  }
+
+  /** @return the result from Musixmatch API */
+  public String grabResponse() throws MalformedURLException, IOException {
+    getResult();
+    return this.response;
+  }
+
+  /**
+   * Builds an appropriate URL with the information given.
+   *
+   * @param operation Musixmatch API method. See above.
+   * @param parameters The parameter string. See above.
+   */
+  private URL buildUrl(String operation, String parameters) throws MalformedURLException {
+    String urlString = baseUrl + operation + outputReqs + parameters + apiKey;
+    URL url = new URL(urlString);
+    return url;
+  }
+
+  /**
+   * Calls the method and gets the result from Musixmatch. Reads data into an input buffer and
+   * returns as string.
```

Comment:
kinda confused about why it needs to go through an input buffer, is this just to store the info before it gets called?

Reply:
Ok, so what's basically happening here is when I call the API it basically returns to me a url to a site where the only contents on it is the response object. What I'm doing here is creating a buffer and reading the contents of that "site" line by line as a stream, and adding the contents to the buffer. 

I don't know the in's and outs about StringBuffer in Java, but to me, this is the equivalent to having an array in C with a buffer of _n_ chars, filling up that array with chars in an input stream, and then mallocing once you get close to the end to allocate more space. The buffer is necessary because we don't know how many characters the response is going to be. 

This link is where I got my info from: https://docs.oracle.com/javase/tutorial/networking/urls/readingURL.html

Reply:
got it thank you